### ISSUE TITLE
Pass number as name in ADKernel 

### ISSUE THREAD

**Issue by tophmatthews** [idaholab/moose#12645]
## Rationale

Traditional kernels allow objects like materials to be set as numbers in the input file or as defaults in the constructers. `ADKernels` do not allow this currently

## Description

You can test how numbers in the input file don't work by changing `ADKernels/diff/ad_mat_prop=1` or `ADKernels/diff/regular_mat_prop=1` in `test/tests/materials/ad_material/ad_material.i.`
Both result in 
```
"Material property '1', requested by 'diff' is not defined on block 0"
```

## Impact

Better usability of `ADKernels`

**Comment by tophmatthews** [comment]
FYI @lindsayad

**Comment by lindsayad** [comment]
Thanks!

**Comment by tophmatthews** [comment]
Solved by #12743
